### PR TITLE
reduce readiness failure threshold

### PR DESCRIPTION
**Pull request by mfrancisc**
Reduce number of errors:

```
{"level":"error","ts":"2024-02-07T02:33:06.081Z","logger":"registration-service","msg":"http: Server closed","timestamp":"Wed, 07 Feb 2024 02:33:06 +0000","commit":"b4a49ca","error":"http: Server closed","stacktrace":"github.com/codeready-toolchain/registration-service/pkg/log.(*Logger).Errorf\n\t/home/runner/work/registration-service/registration-service/pkg/log/log.go:183\ngithub.com/codeready-toolchain/registration-service/pkg/log.(*Logger).Error\n\t/home/runner/work/registration-service/registration-service/pkg/log/log.go:169\ngithub.com/codeready-toolchain/registration-service/pkg/log.Error\n\t/home/runner/work/registration-service/registration-service/pkg/log/log.go:111\nmain.main.func2\n\t/home/runner/work/registration-service/registration-service/cmd/main.go:142"}
```

the readiness probe will fail earlier removing the service from load balancing and thus reducing the number of those errors.

### CODE REVIEW

File: deploy/registration-service.yaml
Comment (alexeykazakov) [inline deploy/registration-service.yaml:117]:
```suggestion
                failureThreshold: 1
```

:)

**Review by alexeykazakov — COMMENTED** [review]
Looks good. But I would set it to 1.

We can't merge it yet though. Let's fix the host operator upgrade first.

**Comment by metlos** [comment]
Please also take into account the discussion on the host-operator PR https://github.com/codeready-toolchain/host-operator/pull/966 that discusses the changes in the registration-service, too.

Sorry for confusing the PRs :)

File: cmd/main.go
Comment (rajivnathan) [inline cmd/main.go:144]: Should we add some additional context to this log? eg. `The server has closed, this is expected when server shutdown has been initiated.`

**Review by rajivnathan — APPROVED** [review]
Nice, I agree it's still good to log something in the case of the `ErrServerClosed` case. Had a small suggestion about the log message to make it a bit more clear why we're logging it as INFO instead of ERROR. But I'll leave it up to you to decide if it's necessary.

File: cmd/main.go
Comment (alexeykazakov) [reply cmd/main.go:144]: +1 @rajivnathan

**Review by alexeykazakov — APPROVED** [review]

File: cmd/main.go
Comment (mfrancisc) [reply cmd/main.go]: makes sense, see [6a877c4](https://github.com/codeready-toolchain/registration-service/pull/401/commits/6a877c4219212c13f1876e8b6d458b4380169797)

thanks for your suggestion!

**Review by alexeykazakov — APPROVED** [review]

**Pull request merged**
